feelpp/feel/feelmesh/bvh.hpp

```diff
@@ -373,9 +379,9 @@ namespace Feel
             }
             else{
                 // Create a node, since there are at least two primitives in the list
-                node->buildInternalNode(current_parent,(cut_dimension+1)%nDim,
-                                    recursiveBuild( node, (cut_dimension+1)%nDim, start_index_primitive, mid, orderedPrims),
-                                    recursiveBuild( node, (cut_dimension+1)%nDim, mid, end_index_primitive, orderedPrims));
+                node->buildInternalNode(current_parent,(cut_dimension+1)%realDim,
```

Comment:
an optimisation will be to precompute only one time this operation (some operation as division, modulo can bea little but expansive). Maybe can be also done at compilation time, no?